apps/namadillo/src/App/Governance/ProposalHeader.tsx

```diff
@@ -211,37 +211,63 @@ const StatusLabel: React.FC<{
 };
 
 const Progress: React.FC<{
-  proposal: AtomWithQueryResult<StoredProposal>;
+  proposal: AtomWithQueryResult<Proposal>;
 }> = ({ proposal }) => {
   return (
     <div className="w-full grid grid-cols-2 text-xs">
       <span>Progress</span>
+      <TimeRemaining proposalQuery={proposal} />
       <div className="col-span-2 mt-3 mb-2">
         <ProgressBar cachedProposal={proposal} />
       </div>
       <ProgressStartEnd
         proposal={proposal}
-        epochKey="startEpoch"
+        timeKey="startTime"
         className="col-start-1"
       />
       <ProgressStartEnd
         proposal={proposal}
-        epochKey="endEpoch"
+        timeKey="endTime"
         className="text-right"
       />
     </div>
   );
 };
 
+const TimeRemaining: React.FC<{
+  proposalQuery: AtomWithQueryResult<Proposal>;
+}> = ({ proposalQuery }) => {
+  const text = (() => {
```

Comment:
Maybe this could be a simple function instead of a self-invoking one

Reply:
I was wondering too about this pattern. I quite like IIFEs for scoping the logic to just the component in the case where it's more complicated than would work with a ternary statement. I can think of a couple of potential drawbacks:

1. It might be slightly less performant (although not sure about this). Something like `useCallback` would probably be slightly better.
2. If we wanted to test just the function logic, we couldn't get access to it since it's inside the component. Defining it as a normal function outside the component would work here, but the function will only be used by the `TimeRemaining` component so I would prefer to scope it to just the component (defining everything in a new file would also allow this, or having an IIFE wrapping the whole component and function).

What do you think?

Reply:
I think useCallback would avoid recreating the function many times, but I think it would be better to make such refactor if we decide to optimize this code for some reason (I don't see it as something necessary now, prefer clarity in this case).

We can also scope the function inside the component, but without being a IIFE. Something like `renderText()` or `getText()`.

Reply:
In that case, I think I will leave this as is just now and if there is a refactor in the future we can look at this again (the same pattern is used in a few of the components in this file).